core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/BulkImport.java

```diff
@@ -576,7 +587,12 @@ private Text toText(byte[] row) {
         Thread.currentThread().interrupt();
         throw new RuntimeException(e);
       } catch (ExecutionException e) {
-        throw new RuntimeException(e);
+        // clean up exception for user
+        Throwable t = e.getCause();
+        if (t instanceof IllegalArgumentException) {
+          throw (IllegalArgumentException) t;
+        } else
+          throw new RuntimeException(t);
```

Comment:
I thought it was cleaner for the user to just drop the ExecutionException coming out of the future since that is just implementation specific.

Reply:
This could lead to a loss of information for someone debugging.  Like they see a stack trace from a background thread in Accumulo, but they don't know anything about the calling code in the foreground thread.

Reply:
I am not sure if the ExecutionException provides valuable information, but definitely want to wrap the IllegalArgumentException.  I would err on the side of caution with omiting the excution exception in that even if does not currently provide anything useful, it may in future versions of Java.

Reply:
Good point.  Better to error on the side of too much information.